app/src/main/java/org/septa/android/app/transitview/TransitViewFragment.java

```diff
@@ -0,0 +1,290 @@
+        this.firstRoute = route;
+        firstRoutePicker.setText(firstRoute.getRouteLongName());
+
+        TransitType transitType = TransitType.BUS;
+        if (isTrolley(firstRoute.getRouteId())) {
+            transitType = TransitType.TROLLEY;
+        }
+
+        // add color bullet icon beside route name
+        int color = ContextCompat.getColor(getContext(), transitType.getLineColor(firstRoute.getRouteId(), getContext()));
+        Drawable bullet = ContextCompat.getDrawable(getContext(), R.drawable.shape_line_marker);
+        bullet.setColorFilter(color, PorterDuff.Mode.SRC);
+        firstRoutePicker.setCompoundDrawablesWithIntrinsicBounds(bullet, null, ContextCompat.getDrawable(getContext(), R.drawable.ic_line_picker), null);
+
+        activateView(secondRoutePicker);
+        activateView(queryButton);
+    }
+
+    @Override
+    public void selectSecondRoute(RouteDirectionModel route) {
+        this.secondRoute = route;
+        secondRoutePicker.setText(secondRoute.getRouteLongName());
+
+        TransitType transitType = TransitType.BUS;
+        if (isTrolley(secondRoute.getRouteId())) {
+            transitType = TransitType.TROLLEY;
+        }
+
+        // add color bullet icon beside route name
+        int color = ContextCompat.getColor(getContext(), transitType.getLineColor(secondRoute.getRouteId(), getContext()));
+        Drawable bullet = ContextCompat.getDrawable(getContext(), R.drawable.shape_line_marker);
+        bullet.setColorFilter(color, PorterDuff.Mode.SRC);
+        secondRoutePicker.setCompoundDrawablesWithIntrinsicBounds(bullet, null, ContextCompat.getDrawable(getContext(), R.drawable.ic_line_picker), null);
+
+        activateView(thirdRoutePicker);
+    }
+
+    @Override
+    public void selectThirdRoute(RouteDirectionModel route) {
+        this.thirdRoute = route;
+        thirdRoutePicker.setText(thirdRoute.getRouteLongName());
+
+        TransitType transitType = TransitType.BUS;
+        if (isTrolley(thirdRoute.getRouteId())) {
+            transitType = TransitType.TROLLEY;
+        }
+
+        // add color bullet icon beside route name
+        int color = ContextCompat.getColor(getContext(), transitType.getLineColor(thirdRoute.getRouteId(), getContext()));
+        Drawable bullet = ContextCompat.getDrawable(getContext(), R.drawable.shape_line_marker);
+        bullet.setColorFilter(color, PorterDuff.Mode.SRC);
+        thirdRoutePicker.setCompoundDrawablesWithIntrinsicBounds(bullet, null, ContextCompat.getDrawable(getContext(), R.drawable.ic_line_picker), null);
+    }
+
+    private void disableView(View view) {
+        view.setAlpha((float) .3);
+        view.setClickable(false);
+    }
+
+    private void activateView(View view) {
+        view.setAlpha(1);
+        view.setClickable(true);
+    }
+
+    private boolean isTrolley(String routeId) {
+        String[] trolleyRouteIds = new String[]{"10", "11", "13", "15", "34", "36", "101", "102"};
```

Comment:
If you are going to hard code these then make them static.  Otherwise you pay the penalty of the creating the array each time you execute this method.  If you make it static it only happens once when the class is loaded.  

Also, think of a way to make this not be hard coded.  Is this in the DB somewhere, can we load it once?  If it has to be hard coded maybe bring it into res/values ?

Reply:
I have it as a single static method in a utils class in the 85 branch, which has the rest of the features implemented. The list of trolley is in the DB in routes_bus with route_type 0. I can refactor to pull the trolley route IDs from the DB, but will also need to put a "fix" around NHSL (which is stored as a trolley).

Reply:
I think the fix around NHSL is fine.  This way no one has to worry when a Trolly line is added to switched to a Bus.  I know that there is a proposal to get rid of a few of the Trolly lines, who knows if SEPTA will reuse the route id for the bus routes that will replace them.

Reply:
Resolved with https://github.com/septadev/SEPTA-Android/pull/182/commits/bfa25af9d50a19225e9d692aebec24e2e3a0c463